### ISSUE TITLE
Change Request: Allow to configure supression file in config

### ISSUE THREAD

**Issue by susnux** [eslint/eslint#20178]
### ESLint version

v9

### What problem do you want to solve?

If you have a custom location for your baseline / suppression file then you always need to configure it on the command line with `--suppressions-location build/eslint-baseline.json`.
This make the scripts much longer and seems to be a missing feature from the suppressions feature.

### What do you think is the correct solution?

Add it to the eslint config like:

```js
// eslint.config.js
export default [
	{
		linterOptions: {
			suppressionsLocation: 'build/eslint-baseline.json',

			// other things like:
			reportUnusedInlineConfigs: 'error',
		},
	},
]
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by kecrily** [comment]
I think this will be helpful, but let's see what others think.

**Comment by cylewaitforit** [comment]
This would be very helpful. 

Right now when working in a monorepo, the bulk suppression files need to be in the individual packages for eslint to pick it up on subsequent runs without explicitly setting the flag on the CLI. 

Our preference would be to configure a single `.eslint-suppressions` directory in the repo root and set that in the root eslint config that all the packages' eslint config inherit from.

**Comment by comp615** [comment]
We also have a monorepo approach with suppressions at each project folder. The problem is that our repo-wide pre-commit hook can't just run eslint anymore:
```
'*': 'eslint --cache --fix --fix-type layout --flag v10_config_lookup_from_file',
```

This approach doesn't seem to find suppressions at all (nor at each project location). CI is fine because we run `nx run-many -t lint`. However, I'd assume that with `v10_config_lookup_from_file`, even running eslint at the root should work as it should appropriately find the suppressions relevant per-file co-located with the corresponding eslint.config? Is that a bug? Does the suppression file location code just need to be merged with the flag? What's the intended setup for monorepo + suppressions?

**Comment by nzakas** [comment]
Thanks for the suggestion. Suprressions are a feature of the CLI, and as such, can't be configured in a config file. 

I'm curious if anyone on this thread can provide a concrete use case. The OP doesn't have a use case and the other comments don't have enough detail to understand how multiple suppression files could be useful. If we can better understand the use case we might be able to come up with a solution.

@comp615 please open a bug report so we can investigate.

**Comment by susnux** [comment]
For our backend we use other linting tools like Psalm where you can specify the baseline in the config and also other configs like report unused baseline entries etc. While you can set all of this on the CLI it gets hard to maintain the scripts instead of just have one source of the truth - the config file.

So currently we just put it into the scripts, but it would be cleaner to just allow setting this in the config.

**Comment by bsangster-figma** [comment]
@nzakas a few thoughts on this as I've been exploring setting this up for our company:
* We also have a monorepo, and adding this to a configuration file would allow us to control it once (in a plugin or other shared file) across the whole monorepo, rather than updating every use of the ESLint CLI to include suppression commands
* Right now, [VSCode is not able to respect suppression files](https://github.com/microsoft/vscode-eslint/issues/1997). I agree with one of the commenters that I believe this to be a _good_ thing, but having this handled via configuration would allow us to have explicit control over how the ESLint extension behaves and ensure that our suppression file is only consumed as part of CI. I'd love to avoid a scenario where the extension starts trying to decide on its own whether to obey suppression files or not.

**Comment by susnux** [comment]
Just to add:
We use a shared config in our ecosystem where every app ideally has the same layout so being able to share those settings would be really helpful for ecosystem consistency.

**Comment by nzakas** [comment]
Thanks for the feedback, everyone. I understand the desire to put this into a configuration file, but as I already stated, that's not the way ESLint is architected. Our config files are meant to be runtime agnostic, meaning they don't know anything about the file system. Plus, all configs are designed to be applied to a subset of files, which means we could end up needing to write to multiple suppressions files for a single run, and the complexity of making that work in a logical way is substantial. It's the CLI that has knowledge of the file system which is why the suppressions feature lives there and I don't see a scenario where we'd want to support multiple suppression files for a single run.

If the goal is to configure a single suppressions file for a monorepo, I think that can be accomplished with a wrapper script that reads the location from a file of your choice.

**Comment by bsangster-figma** [comment]
Alright, that seems like pretty reasonable pushback. I guess I was hoping that this would be more akin to something like how there are [helpers for consuming `.gitignore` files](https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files), but I can see how that could be trickier for writing to suppressions files rather than purely consuming.

**Comment by comp615** [comment]
@nzakas would it be possible to provide more guidance or docs on what this _should_ look like a monorepo use case? We had this bug which closed https://github.com/eslint/eslint/issues/19954. But I hear your feedback and maybe it's something we need to run differently. The two cases we're trying to reconcile are:
1) Each project has it's own lint target...these are used in CI
2) As a convenience in IDEs or pre-commit hooks (similar to my code example), we run lint only on changed files from the root. 

Determining and then running each package lint individually in the precommit is slower and clunky to the point of needing to remove lint from pre-commit altogether. Or people just get all the suppressed errors in pre-commit. So that's why I'm pushing so hard; it worked prior to these because in effect, running `nx run package:lint` was the same as `eslint pacakge/file` (i.e. the configs get applied consistently)

**Comment by nzakas** [comment]
@comp615 it's really difficult to provide guidance without seeing a concrete example. I do offer paid video consultations to help folks go through their specific use cases without needing to disclose anything on public repos. Please feel free to reach out if you're interested.
